project_task_schedule/models/project_task.py

```diff
@@ -0,0 +1,25 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    enable_task_planning = fields.Boolean(
+        related='project_id.enable_task_planning')
+    date_range_id = fields.Many2one(
+        comodel_name="date.range", string="Date range")
+    start_date = fields.Date("Start Date")
```

Comment:
Maybe this needs a default of today, as well as end date, so that there is always a value in it after it's created for example on timeline